app/Http/Controllers/API/UserController.php

```diff
@@ -0,0 +1,273 @@
+
+        if ($paramIncludeDivisionUsers) {
+            $returnUsers = User::where('subdivision', config('app.owner_short'));
+            if ($paramOnlyActive) {
+                $returnUsers = $returnUsers->where('atc_active', true);
+            }
+            $returnUsers = $returnUsers->with($queryInclude)->get();
+        }
+
+        if ($paramIncludeEndorsements) {
+            $endorsementUsers = User::whereHas('endorsements', function (Builder $q) {
+                $q->where('expired', false)->where('revoked', false);
+            })->whereNotIn('id', $returnUsers->pluck('id'));
+            if ($paramOnlyActive) {
+                $endorsementUsers = $endorsementUsers->where('atc_active', true);
+            }
+            $endorsementUsers = $endorsementUsers->with($queryInclude)->get();
+
+            $returnUsers = $returnUsers->merge($endorsementUsers);
+        }
+
+        if ($paramIncludeRoles) {
+            $roleUsers = User::whereHas('groups')->whereNotIn('id', $returnUsers->pluck('id'));
+            if ($paramOnlyActive) {
+                $roleUsers = $roleUsers->where('atc_active', true);
+            }
+            $roleUsers = $roleUsers->with($queryInclude)->get();
+
+            $returnUsers = $returnUsers->merge($roleUsers);
+        }
+
+        if ($paramIncludeTraining) {
+            $trainingUsers = User::whereHas('trainings', function (Builder $q) {
+                $q->where('status', '>=', 0);
+            })->whereNotIn('id', $returnUsers->pluck('id'));
+            if ($paramOnlyActive) {
+                $trainingUsers = $trainingUsers->where('atc_active', true);
+            }
+            $trainingUsers = $trainingUsers->with($queryInclude)->get();
+
+            $returnUsers = $returnUsers->merge($trainingUsers);
+        }
+
+        //
+        // Process the paramters
+        //
+
+        // Endorsements
+        if ($paramIncludeEndorsements) {
+            foreach ($returnUsers as $user) {
+                $user->endorsements = $this->mapEndorsements($user->endorsements->where('expired', false)->where('revoked', false));
+            }
+        }
+
+        // Roles
+        if ($paramIncludeRoles) {
+            foreach ($returnUsers as $user) {
+                $user->roles = collect();
+
+                foreach (Area::all() as $area) {
+                    $areaRoles = $user->groups->where('pivot.area_id', $area->id)->pluck('name');
+
+                    if ($areaRoles->count()) {
+                        $user->roles[$area->name] = ($user->groups->where('pivot.area_id', $area->id)->pluck('name'));
+                    } else {
+                        $user->roles[$area->name] = null;
+                    }
+                }
+            }
+        }
+
+        // Trainings
+        if ($paramIncludeTraining) {
+            foreach ($returnUsers as $user) {
+                $user->training = $this->mapTrainings($user->trainings->where('status', '>=', 0));
+            }
+        }
```

Comment:
each of these are essentially functions that expand the current data with additional information. Instead of adding a bunch of ifs, you could create a table lookup, where you get the distinctive options that are set and execute each of the matches.

Alternatively, you could extract some of these as functions that exist on the model, and then re-use them here.

Reply:
>  where you get the distinctive options that are set and execute each of the matches.
Not sure I misunderstand, but this is what I'm doing currently. I'm executing each of the matches if the parameter is present? If you could write a sample snip of what you mean if I misunderstood